### PR TITLE
containerized development environment in a docker so that it's most consistent and onboarding is easier

### PR DESCRIPTION
**Pull request by abdullahalhoothy**

### CODE REVIEW

**Comment by mattlindsey** [comment]
Hi. Small comment.  You may want to call README_old.MD something else, like README_local_setup.MD.  Or put the local setup instructions somewhere else since not everyone may want to use Docker.  :)

**Comment by abdullahalhoothy** [comment]
Thanks matt, the goal is not to make it complicated and i understand some don't want to work with docker. for the time being i added the local development method to the same readme. i will delete the old readme , i forgot about it

**Comment by mattlindsey** [comment]
Goal sounds good Abdullah, although I'm a little worried that if this PR is merged the instructions won't contain some useful information for 'local' developers, which I think you deleted.  If I'm wrong and they are somewhere, don't worry.

**Comment by abdullahalhoothy** [comment]
yes i deleted it because it was alot of information while useful wasn't entirely about setup but rather file structure and dependencies, which are managed by npm anyway 

i created a new file called setup dependencies with those information in there, i will move that into a doc directory

File: .gitignore [hunk 1]
```diff
@@ -45,15 +45,13 @@ firestore-data
...
-.firebaserc
 **/*irebase-config*
 **/*irebaseConfig*
-**/firebase.json
-**/*.rules
 **/*secret*
-**/*indexes*
 **/*adminsdk*
 **/*service*
+firebaseConfig.json   # Added line
```
Comment (mattlindsey) [inline .gitignore:53]: `# Added line` comment unnecessary

File: Setup dependencies.md
Comment (mattlindsey) [inline Setup dependencies.md:1]: Maybe move this file to a `docs` directory named `setup_notes.md`

File: docker_image_build_files/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# RUN nvm install 18.19.0
+# RUN nvm use 18.19.0
+RUN mkdir -p /home/user/projects/
+WORKDIR /home/user/projects/
+
+
+# # Clone GitHub repository
```
Comment (mattlindsey) [inline docker_image_build_files/Dockerfile:16]: Is this normal to clone and install git?  I haven't noticed it elsewhere.

File: docker_image_build_files/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,100 @@
+# a Node.js base image that includes Node v18
+FROM node:18-bullseye
+
+# Install Git and Java JDK and jq
+RUN apt-get update && \
+    apt-get install -y git default-jdk jq && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# RUN nvm install 18.19.0
```
Comment (mattlindsey) [inline docker_image_build_files/Dockerfile:10]: Remove commented lines

**Review by mattlindsey — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -4,317 +4,101 @@
...
 ```
-{
-    "indexes": [],
-    "fieldOverrides": []
-}
+13. see the output you can run in your host machine browser http://localhost:3000
+
+
+## Dev Local Setup
+Below are linux command used to setup the environment
```
Comment (mattlindsey) [inline README.md:37]: Add a comment that notes for setup on Mac are in `docs\setup_notes.md`, or a short list of the Mac commands (like `brew`) here.

**Review by mattlindsey — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (abdullahalhoothy) [reply .gitignore:53]: True

File: docker_image_build_files/Dockerfile (hunk 2 above)
Comment (abdullahalhoothy) [reply docker_image_build_files/Dockerfile:16]: The Ideal would be to use Attachment Volumes, but that would make it more complicated i feel
Cloning to git works especially if just adopt working with remote containers , that way we encourage committing early and often

File: README.md (hunk 4 above)
Comment (abdullahalhoothy) [reply README.md:37]: i'm going to ask chatgpt to give the equivalent commands for mac using brew , and i will put that in the read me

Comment (abdullahalhoothy) [reply README.md:37]: or since i can't verify it , ill just say that notes for setup on Mac are in docs\setup_notes.md

**Pull request merged**